deploy/controllers/BankerJoeController.ts

```diff
@@ -0,0 +1,20 @@
+import { HardhatRuntimeEnvironment } from 'hardhat/types';
+import { DeployFunction } from 'hardhat-deploy/types';
+import { BankerJoeControllerV4__factory } from '../../typechain';
+import { getControllerArgs } from './args';
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
```

Comment:
very similar function in various places, would be possible to create a `factory function` where it can take the tags as params and reduce the amount of repeated similar code?

Reply:
My thought process: This is a simple deploy and would only be abstracting boiler plate code felt like over engineering it. If abstracting I would like to make the named account (deployer) configurable through the parameters which gets a little tricky due to how the `getNamedAccounts` imports.

Assume, all deployers are the same:
```
const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
    const contract_name = (new AaveControllerV4__factory()).contractName;
    const deployment_name = contract_name; 
    await deployController(hre, contract_name, deployment_name);
};
```

If deployer wants to be configurable then:
```    
const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
    const {deployments, getNamedAccounts} = hre;
    const {deploy} = deployments;
    const {differentDeployer} = await getNamedAccounts();
    const contract_name = (new AaveControllerV4__factory()).contractName;
    const deployment_name = contract_name;
    await deployController(hre, contract_name, deployment_name, differentDeployer);
};
```
which doesnt save much, altho named deployer assumptions are probably fine.

Reply:
still generalized it tho